### PR TITLE
Update SSH docs 

### PR DESCRIPTION
**Pull request by aclevername**
### Description

#3198

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: userdocs/src/usage/managing-nodegroups.md [hunk 1]
```diff
@@ -160,6 +160,35 @@ There are no specific commands in `eksctl`to update the labels of a nodegroup bu
 kubectl label nodes -l alpha.eksctl.io/nodegroup-name=ng-1 new-label=foo
 ```
 
+### SSH Access
```
Comment (aclevername) [inline userdocs/src/usage/managing-nodegroups.md:163]: We don't really have a great place for these docs to live, and I don't really want to add a page for just SSH configuration. I've put it here for now, open to suggests

File: pkg/apis/eksctl.io/v1alpha5/types.go [hunk 2]
```diff
@@ -1058,20 +1058,22 @@ type (
...
-		// +optional
+		// +optional Public key name in EC2 to be added to the nodes SSH keychain. If Allow is false this value
+		// is ignored.
 		PublicKeyName *string `json:"publicKeyName,omitempty"`
 		// +optional
 		SourceSecurityGroupIDs []string `json:"sourceSecurityGroupIds,omitempty"`
 		// Enables the ability to [SSH onto nodes using SSM](/introduction#ssh-access)
 		// +optional
-		EnableSSM *bool `json:"enableSsm,omitempty"`
+		EnableSSM *bool `json:"enableSSM,omitempty"`
```
Comment (michaelbeaumont) [inline pkg/apis/eksctl.io/v1alpha5/types.go:1076]: hmm this is a breaking change unfortunately... ugh the `json` tags in go are such a terrible, inflexible abstraction

Comment (aclevername) [reply pkg/apis/eksctl.io/v1alpha5/types.go:1076]: wait really?! I always thought json wasn't case sensitive for key names 😢  I'll change it back...

Comment (michaelbeaumont) [reply pkg/apis/eksctl.io/v1alpha5/types.go]: That'd be nice, check https://github.com/weaveworks/eksctl/issues/753

File: userdocs/src/usage/managing-nodegroups.md [hunk 3]
```diff
@@ -160,6 +160,35 @@ There are no specific commands in `eksctl`to update the labels of a nodegroup bu
...
+You can enable SSH access for nodegroups by configuring one of `publicKey`, `publicKeyName` and `publicKeyPath` in your
```
Comment (Callisto13) [inline userdocs/src/usage/managing-nodegroups.md:164]:
```suggestion
You can enable SSH access for nodegroups by configuring one of `publicKey`, `publicKeyName` or `publicKeyPath` in your
```

**Review by Callisto13 — APPROVED** [review]

File: pkg/apis/eksctl.io/v1alpha5/types.go [hunk 4]
```diff
@@ -1058,14 +1058,16 @@ type (
...
-		// are ignored if set to false
+		// +optional If Allow is true the SSH configuration provided is used, otherwise it is ignored. Only one of
+		// PublicKeyPath, PublicKey and PublicKeyName can be configured
 		Allow *bool `json:"allow"`
-		// +optional
+		// +optional The path to the SSH public key to be added to the nodes SSH keychain. If Allow is true this value
+		// defaults to "~/.ssh/id_rsa.pub", otherwise the value is ignored.
 		PublicKeyPath *string `json:"publicKeyPath,omitempty"`
-		// +optional
+		// +optional Public key to be added to the nodes SSH keychain. If Allow is false this value is ignored.
```
Comment (michaelbeaumont) [inline pkg/apis/eksctl.io/v1alpha5/types.go:1067]: I'm not sure every field needs to have "If Allow is false this value is ignored." It looks pretty noisy at https://deploy-preview-3233--eksctl.netlify.app/usage/schema/#nodeGroups-ssh-allow

File: pkg/apis/eksctl.io/v1alpha5/types.go [hunk 5]
```diff
@@ -1058,14 +1058,16 @@ type (
...
 	// NodeGroupSSH holds all the ssh access configuration to a NodeGroup
 	NodeGroupSSH struct {
-		// +optional Enables/Disables the security group configuration. Values provided by SourceSecurityGroupIDs
-		// are ignored if set to false
+		// +optional If Allow is true the SSH configuration provided is used, otherwise it is ignored. Only one of
+		// PublicKeyPath, PublicKey and PublicKeyName can be configured
 		Allow *bool `json:"allow"`
-		// +optional
+		// +optional The path to the SSH public key to be added to the nodes SSH keychain. If Allow is true this value
+		// defaults to "~/.ssh/id_rsa.pub", otherwise the value is ignored.
```
Comment (michaelbeaumont) [inline pkg/apis/eksctl.io/v1alpha5/types.go:1065]:
```suggestion
		// The path to the SSH public key to allow access to the nodes
		// Defaults to `"~/.ssh/id_rsa.pub"`
```

**Review by michaelbeaumont — APPROVED** [review]
@aclevername I shortened the strings further (looks better on the schema page), removed the `+optional` tags hanging around which don't do anything and rebased!

**Comment by aclevername** [comment]
> @aclevername I shortened the strings further (looks better on the schema page), removed the `+optional` tags hanging around which don't do anything and rebased!

Thanks!

**Pull request merged**